agent/util-scripts/pbench-tool-meister-start

```diff
@@ -244,13 +204,635 @@ def waitpid(pid):
     return exit_status
 
 
-def main(_prog, cli_params):
-    """Main program for the tool meister start.
+class ReturnCodeErr(RuntimeError):
+    """Err - exception definition to capture return code as an attribute.
+    """
+
+    def __init__(self, message, return_code):
+        super().__init__(message)
+        self.return_code = return_code
+
+
+class StartTmsErr(ReturnCodeErr):
+    pass
+
+
+def start_tms_via_ssh(
+    exec_dir, tool_group, ssh_opts, full_hostname, redis_server, redis_client, logger
+):
+    """start_tms_via_ssh - orchestrate the creation of local and remote Tool
+    Meister instances using ssh for those that are remote.
+
+    Raises a StartTmsErr on failure.
+
+    NOTE: all local and remote Tool Meisters are started even if failures
+    occur for some; this allows the user to see logs for all the individual
+    failures.
+    """
+    failures = 0
+    successes = 0
+    # NOTE: it is assumed that the location of the pbench-tool-meister command
+    # is the same on the local host as it is on any remote host.
+    tool_meister_cmd = exec_dir / "tool-meister" / "pbench-tool-meister"
+    ssh_cmd = "ssh"
+    ssh_path = find_executable(ssh_cmd)
+    base_args = [
+        ssh_cmd,
+    ]
+    base_args.extend(shlex.split(ssh_opts))
+    args = [
+        "<host replace me>",
+        f"{tool_meister_cmd}-remote",
+        redis_server.host,
+        str(redis_server.port),
+        "<tm param key>",
+        "yes",  # Yes, request the tool meister daemonize itself
+    ]
+    tms = dict()
+    tm_count = 0
+    for host in tool_group.hostnames.keys():
+        tm_count += 1
+        tm_param_key = f"tm-{tool_group.group}-{host}"
+        if host == full_hostname:
+            logger.debug("5a. starting localhost tool meister")
+            try:
+                pid = os.fork()
```

Comment:
First, I don't really want to change this process in this commit, as this code is really just a "lift-n-shift" to move the behavior to a new method.

As for the history, this was a very early decision made in the Tool Meister development where it seemed to gain us no real advantage since we still have to wait for the combined set of processes, and the `subprocess` interfaces don't ease the logic for that.

Reply:
Re "lift-n-shift", got it!

As for the rest of it, I think I disagree, but I'm not inclined to debate it in this PR.